onnxruntime/core/session/abi_session_options.cc

```diff
@@ -179,3 +179,9 @@ ORT_API_STATUS_IMPL(OrtApis::DisablePerSessionThreads, _In_ OrtSessionOptions* o
   options->value.use_per_session_threads = false;
   return nullptr;
 }
+
+ORT_API_STATUS_IMPL(OrtApis::SetSessionConfiguration, _Inout_ OrtSessionOptions* options,
+                    _In_ OrtSessionConfigKey config_key, _In_ const char* config_value) {
+  options->value.session_configurations[config_key] = config_value;
```

Comment:
Do we need to verify if the config_key is valid and limit the length of the config_value here (say 32 chars max)?

Reply:
Would be nice to check key is valid but it doesn't really hurt if it isn't. May need something like this though: [https://stackoverflow.com/questions/4165439/generic-way-to-cast-int-to-enum-in-c](https://stackoverflow.com/questions/4165439/generic-way-to-cast-int-to-enum-in-c)

32 sounds too short, but we should have a max (and for security reasons we should be checking the value doesn't exceed that).

I'd suggest 1024 chars. Could have a long filename or maybe multiple filenames. Could have a small amount of base64 encoded binary data. 

---
In reply to: [466157824](https://github.com/microsoft/onnxruntime/pull/4718#discussion_r466157824) [](ancestors = 466157824)